src/lib.rs

```diff
@@ -302,7 +308,8 @@ impl ClientBuilder {
             .etherscan_urls()
             .map(urls)
             .ok_or_else(|| EtherscanError::ChainNotSupported(chain))?;
-        self.with_api_url(etherscan_api_url?)?.with_url(etherscan_url?)
+
+        self.with_chain_id(chain.id()).with_api_url(etherscan_api_url?)?.with_url(etherscan_url?)
```

Comment:
we need to add a note to function docs that this also sets the chain id for multichain verification